File: PianoKeyboard_java_eclipse/src/com/teuskim/pianokeyboard/WhiteKeyView.java
package com.teuskim.pianokeyboard;

import android.content.Context;
import android.util.AttributeSet;

public class WhiteKeyView extends KeyView {

	public WhiteKeyView(Context context, AttributeSet attrs, int defStyle) {
		super(context, attrs, defStyle);
	}

	public WhiteKeyView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public WhiteKeyView(Context context) {
		super(context);
	}

	@Override
	protected int getLayoutResId() {
		return R.layout.key_white;
	}

	@Override
	protected int getNormalImageResId() {
		return R.drawable.key_white;
	}

	@Override
	protected int getPressedImageResId() {
		return R.drawable.key_white_pressed;
	}

	@Override
	public int getKeyType() {
		return KEY_TYPE_WHITE;
	}

}

File: PianoKeyboard_kotlin/app/src/main/java/com/teuskim/pianokeyboard/WhiteKeyView.kt
package com.teuskim.pianokeyboard

import android.content.Context
import android.util.AttributeSet

class WhiteKeyView : KeyView {

    protected override val layoutResId: Int
        get() = R.layout.key_white

    protected override val normalImageResId: Int
        get() = R.drawable.key_white

    protected override val pressedImageResId: Int
        get() = R.drawable.key_white_pressed

    override val keyType: Int
        get() = KeyView.KEY_TYPE_WHITE

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context) : super(context) {}

}
